src/lightning/pytorch/trainer/connectors/accelerator_connector.py

```diff
@@ -458,12 +458,9 @@ def _choose_strategy(self) -> Union[Strategy, str]:
                 device = "cpu"
             # TODO: lazy initialized device, then here could be self._strategy_flag = "single_device"
             return SingleDeviceStrategy(device=device)  # type: ignore
-        if len(self._parallel_devices) > 1:
-            if _IS_INTERACTIVE:
-                return "ddp_fork"
-            return "ddp_spawn"
-
-        return DDPStrategy.strategy_name
+        if len(self._parallel_devices) > 1 and _IS_INTERACTIVE:
+            return "ddp_fork"
```

Comment:
in past we had a few silly bugs with typos, so how about using enums internally?

Reply:
We had enums in the past and we decided to remove them for better readability. Also, since then we introduced the strategy registry. We have many tests that, so if there was a typo, they would fail.